From: Stefan Haberland <stefan.haberland@de.ibm.com>
Subject: dasd: Fix unresumed device after suspend/resume.
Patch-mainline: not yet
Git-commit: -
References: bnc#927308, LTC#123892

Description:  dasd: Fix unresumed device after suspend/resume.
Symptom:      DASD devices are not responding after a suspend
              resume cycle.
Problem:      The DASD device driver only has a limited amount of
              memory to build I/O requests.
              This memory was used by blocklayer requests leading to
              an inability to build needed internal requests to resume
              the device.
Solution:     Fix by preventing the DASD driver to fetch requests for
              a stopped device.
Reproduction: Do suspend/resume cycles with DASD devices attached and
              I/O load.

Signed-off-by: Stefan Haberland <stefan.haberland@de.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/block/dasd.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/s390/block/dasd.c
+++ b/drivers/s390/block/dasd.c
@@ -2539,6 +2539,11 @@ static void __dasd_process_request_queue
 			__blk_end_request_all(req, -EIO);
 		return;
 	}
+
+	/* if device ist stopped do not fetch new requests */
+	if (basedev->stopped)
+		return;
+
 	/* Now we try to fetch requests from the request queue */
 	while ((req = blk_peek_request(queue))) {
 		if (basedev->features & DASD_FEATURE_READONLY &&
